data-tck/src/main/groovy/io/micronaut/data/tck/tests/AbstractRepositorySpec.groovy

```diff
@@ -2554,6 +2560,10 @@ abstract class AbstractRepositorySpec extends Specification {
         loadedBook2
         loadedBook2.title == book2.title
         loadedBook2.id == book2.id
+        when:
+        students = studentRepository.queryByName(student.name, Pageable.from(0, 2))
+        then:
+        students.size() == 1
```

Comment:
This returns 2 records but with the same student (because left joined to two books)